### PR TITLE
Dev UX improvements (Tenderly + env cleanup + config cleanup)

### LINKED ISSUE #261: Integrate tenderly into repo
@SkandaBhat

### PR DESCRIPTION
**Pull request by SkandaBhat**
- Integrated tenderly into the project. We can automatically track deployments on tenderly dashboard (https://dashboard.tenderly.co/razor/razor-network/contracts)
- Removed ganache throughout. We can use `hardhat run node` to run a chain locally.
- Improved deploy scripts nomenclature in package.json. We can now use `npm run deploy:local` and `npm run deploy:mumbai` respectively
- Cleaned up unnecessary jsons and abis
- Updated readme
- Bumped package.json version to latest
- Cleaned up deploy.sh

Fixes #261

### CODE REVIEW

**Comment by SkandaBhat** [comment]
Tests are failing. Do not merge.

File: hardhat.config.js [hunk 1]
```diff
@@ -41,29 +42,17 @@ module.exports = {
...
       gas: 7700000,
-      gasPrice: 1 * GWEI,
-    },
-    matic_mumbai_testnet: {
-      url: PROVIDER_URL || '',
-      accounts: { mnemonic: MNEMONIC },
-      chainId: ENV_CHAIN_IDS[NETWORK],
-      gas: 7700000,
-      gasPrice: 1 * GWEI,
+      gasPrice: 30 * GWEI,
```
Comment (hrishikeshio) [inline hardhat.config.js:55]: do we need to hardcode gasprice?

Comment (SkandaBhat) [reply hardhat.config.js:55]: > gasPrice: Its value should be "auto" or a number. This parameter behaves like gas. Default value: "auto".

Lets set it to 'auto'.

Comment (hrishikeshio) [reply hardhat.config.js:55]: yes

Comment (SkandaBhat) [reply hardhat.config.js]: I've removed `gas` and `gasPrice`. Hardhat will set these by default to 'auto'.

**Comment by SkandaBhat** [comment]
Tests are passing now @hrishikeshio you can merge

**Pull request merged**